source/level.js

```diff
@@ -201,44 +229,67 @@ export class Level extends Phaser.Scene{
         // Turn characterData into a json string and store it in a cookie
         let jsonString = JSON.stringify(this.characterData);
         document.cookie = "characterData=" + jsonString + ";" + expireString + ";path=/;";
-
     }
+
     addGold(addedGold){
         this.characterData.gold += addedGold;
         this.goldText.text = 'Gold: ' + this.characterData.gold;
     }
-    enemyKilled(){
+
+    enemyKilled(name){
         // Update kill quest score
-        if (this.characterData[this.background.name].enemiesKilled < this.killQuest) {
+        if (this.characterData[this.background.name].enemiesKilled[name] < this.killQuest) {
             this.characterData.totalEnemiesKilled++;
-            this.characterData[this.background.name].enemiesKilled++;
-            this.killQuestText.text = this.characterData[this.background.name].enemiesKilled + "/" + this.killQuest + " " + this.enemySettings.name + "s killed";
+            this.characterData[this.background.name].enemiesKilled[name]++;
+
+            let questCompleted = true;
```

Comment:
I would only set this to true in the if statement down that checks for completion.

Reply:
This is part of that statement. I set this local variable to true then I loop through all of the enemies and switch it to false if any do not meet it. Then I check that Boolean value

Reply:
Yeah I was saying that more for readability. On second thought, just get rid of the variable and do the stuff that is in if (quest complete) inside the if before it (switch the < to >=)

Reply:
But it needs to come afterwards so you know that all of the enemies are defeated X # of times. For example 10 cows and 10 goblins defeated. Otherwise it would say completed when only one of them has been. I guess this is an intended functionality question

Reply:
We can leave it as is if you want - it is really more of a nit pick. I just don't like assuming it to be true until all the monster counts are proven to be >=.